### PR TITLE
Milestone: Decorate a class

### PR DESCRIPTION
**Pull request by Kashif-Saleem-Ghuman**
In this milestone 💪 

- [ ] Interface
  - [ ] Created a class Nameable.
    - [ ] Implemented a method called correct_name that will raise a NotImplementedError.
- [ ] Turned Person class to Nameable
  - [ ] Person class inherits from Nameable
  - [ ] this class has a method correct_name implemented. It simply returns the name attribute.
- [ ] Prepared base Decorator
  - [ ] It inherits from Nameable.
  - [ ] In the constructor assigned a nameable object from params to an instance variable.
  - [ ] Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
- [ ] Prepared CapitalizeDecorator and TrimmerDecorator
  - [ ] For the CapitalizeDecorator:
    - [ ] Created a class that inherits from the base Decorator class.
    - [ ] Implemented a method correct_name that capitalizes the output of @nameable.correct_name.
- [ ] For the TrimmerDecorator:
  - [ ] Created a class that inherits from the base Decorator class.
  - [ ] Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
```
Comment (OLIPLICHE) [inline capitalize_decorator.rb:5]: - [ ]  Hi, Here is the outcome I am getting when checking your codes.  Please can you check and see if you can fix it?

Capture

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
```
Comment (OLIPLICHE) [inline decorator.rb:4]: - [ ] I would advise you to add this line `attr_accessor :nameable`, This property will automatically add getter and setter to the method you created.

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Kashif-Saleem-Ghuman , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit has a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kashif-Saleem-Ghuman** [comment]
@OLIPLICHE Thank you for reviewing my PR
I do not understand the issue. When I run the "main.rb" file, it works perfectly fine
asdasdasdasdasdasdasdasdasdasd

**Comment by OLIPLICHE** [comment]
> @OLIPLICHE Thank you for reviewing my PR I do not understand the issue. When I run the "main.rb" file, it works perfectly fine asdasdasdasdasdasdasdasdasdasd

@Kashif-Saleem-Ghuman, Thank you for your feedback. I saw my mistake. I did not require Trimmer... and Capitalize..., I forgot. I am sorry. Thank you for reaching out 👍 
- **Please submit it again, the next reviewer will approve it.**

**Comment by Kashif-Saleem-Ghuman** [comment]
@OLIPLICHE No problem!

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kashif-Saleem-Ghuman** [comment]
@ichala Thank you for approving my PR

**Pull request merged**